gradient_sim.py

```diff
@@ -137,346 +149,307 @@
+            mat=MAT[climnum]))
+
+    Ndemand = total_inputs/litter_CN_site
+    # Calculate plant Ndemand from N in litter production
+    # Assuming plant N_litter balances plant N_uptake and plant not relying on roots
+    # This will not be needed once the model is coupled with a plant model
+
+    Ctransfer['ECM'] = total_inputs * myc_ratio_litter * ECM_pct[plotnum] / 100
+    Ctransfer['AM'] = total_inputs * myc_ratio_litter * (1-ECM_pct[plotnum] / 100)
+
+    result = CORPSE_integrate.run_CORPSE_ODE(T=MAT[climnum], theta=theta, Ndemand=Ndemand,
+                                             inputs=dict([(k, inputs[k][plotnum]) for k in inputs]),
+                                             clay=clay[claynum], initvals=SOM_init, params=params,
+                                             times=spinuptimes, runtype='Spinup')
+    result = CORPSE_integrate.run_CORPSE_ODE(T=MAT[climnum], theta=theta, Ndemand=Ndemand,
+                                             inputs=dict([(k, inputs[k][plotnum]) for k in inputs]),
+                                             clay=clay[claynum], initvals=result.iloc[-1], params=params,
+                                             times=finaltimes, runtype='Final')
+    SAPCarray = result['SAPC']
+    ECMCarray = result['ECMC']
+    AMCarray = result['AMC']
+    UnFastCCarray = result['uFastC']
+    UnSlowCCarray = result['uSlowC']
+    UnNecroCCarray = result['uNecroC']
+    UnprotectedCarray = sumCtypes(result.iloc[:], 'u', 'C')
+    PFastCCarray = result['pFastC']
+    PSlowCCarray = result['pSlowC']
+    PNecroCCarray = result['pNecroC']
+    ProtectedCarray = sumCtypes(result.iloc[:], 'p', 'C')
+    InorgNarray = result['inorganicN']
+    IntNarray = result['Int_N']
+    TotNarray = sumCtypes(result.iloc[:], 'u', 'N') + sumCtypes(result.iloc[:], 'p', 'N')
+    NfromNecroarray = result['NfromNecro']
+    NfromSOMarray = result['NfromSOM']
+    Nlimitarray = result['Nlimit']
+    for timenum in range(timesteps):
+        filename = str(nclimates*nclays*plotnum+nclimates*claynum + climnum + 1) + "_Quarterly_data.txt"
+        if timenum == 0:
+            f = open(filename, "w")
+        else:
+            f = open(filename, "a")
+        f.write("{:.6f} {:.6f} {:.6f} {:.6f} {:.6f} {:.6f} {:.6f} {:.6f} {:.6f}\n".format(InorgNarray[timenum],
+                                                                                          SAPCarray[timenum],
+                                                                            UnprotectedCarray[timenum], ECMCarray[timenum],
+                                                                            AMCarray[timenum],
+                                                                            TotNarray[timenum], NfromNecroarray[timenum],
+                                                                            NfromSOMarray[timenum],Nlimitarray[timenum]))
+    f.close()
```

Comment:
This is a pretty inefficient way to write out the file. Because `run_CORPSE_ODE` returns a pandas dataframe, you could do all of this in one line with `result.to_csv(filename)` (with whatever options you want).

Reply:
Thanks very much for the suggestion! I have simplified it by using the "result.to_csv" as you suggested